forui/lib/src/widgets/line_calendar.dart

```diff
@@ -0,0 +1,352 @@
+                children: [
+                  // TODO: replace with DefaultTextStyle.merge when textHeightBehavior has been added.
+                  merge(
+                    textHeightBehavior: const TextHeightBehavior(
+                      applyHeightToFirstAscent: false,
+                      applyHeightToLastDescent: false,
+                    ),
+                    style: selected ? style.selectedDateTextStyle : style.unselectedDateTextStyle,
+                    child: Text(date.day.toString()),
+                  ),
+                  const SizedBox(height: _textSpacing),
+                  // TODO: replace with DefaultTextStyle.merge when textHeightBehavior has been added.
+                  merge(
+                    textHeightBehavior: const TextHeightBehavior(
+                      applyHeightToFirstAscent: false,
+                      applyHeightToLastDescent: false,
+                    ),
+                    style: selected ? style.selectedDayTextStyle : style.unselectedDayTextStyle,
+                    child: Text(_days[date.weekday - 1]),
+                  ),
+                ],
+              ),
+            ),
+          ),
+          Positioned(
+            top: 6,
+            right: 6,
+            child: Container(
+              height: 4,
+              width: 4,
+              decoration: BoxDecoration(
+                color: _style(context, selected),
+                shape: BoxShape.circle,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('selected', selected))
+      ..add(DiagnosticsProperty('date', date))
+      ..add(DiagnosticsProperty('underline', today));
+  }
+}
+
+/// [FAvatar]'s style.
+final class FLineCalendarStyle with Diagnosticable {
+  /// The vertical padding around the text in the calendar items.
+  final double heightPadding;
+
+  /// The horizontal padding around the text in the calendar items.
+  final double itemPadding;
+
+  /// The box decoration for a selected date.
+  final BoxDecoration selectedDecoration;
+
+  /// The box decoration for an unselected date.
+  final BoxDecoration unselectedDecoration;
+
+  /// The color of the indicator for the selected current date.
+  final Color selectedCurrentDateIndicatorColor;
+
+  /// The color of the indicator for the unselected current date.
+  final Color unselectedCurrentDateIndicatorColor;
+
+  /// The text style for the selected date.
+  final TextStyle selectedDateTextStyle;
+
+  /// The text style for the unselected date.
+  final TextStyle unselectedDateTextStyle;
+
+  /// The text style for the selected day of the week.
+  final TextStyle selectedDayTextStyle;
+
+  /// The text style for the unselected day of the week.
+  final TextStyle unselectedDayTextStyle;
```

Comment:
I think it'll be better if this was abstracted out. You could use `FCalendarStyle`'s state stuff as a reference.